### DISCUSSION TITLE
2020-10-27 feedback triage meeting

### DISCUSSION THREAD

**Discussion by MylesBorins** [npm/feedback#69] [General]
## Time

**UTC Tue 27-Oct-2020 14:00 (02:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Tue 29-Oct-2020 07:00 (07:00 AM) |
| US / Mountain | Tue 29-Oct-2020 08:00 (08:00 AM) |
| US / Central  | Tue 29-Oct-2020 09:00 (09:00 AM) |
| US / Eastern  | Tue 29-Oct-2020 10:00 (10:00 AM) |
| EU / Western  | Tue 29-Oct-2020 14:00 (02:00 PM) |
| EU / Central  | Tue 29-Oct-2020 15:00 (03:00 PM) |
| EU / Eastern  | Tue 29-Oct-2020 16:00 (04:00 PM) |
| Moscow        | Tue 29-Oct-2020 17:00 (05:00 PM) |
| Chennai       | Tue 29-Oct-2020 19:30 (07:30 PM) |
| Hangzhou      | Tue 29-Oct-2020 22:00 (10:00 PM) |
| Tokyo         | Tue 29-Oct-2020 23:00 (11:00 PM) |
| Sydney        | Wed 30-Oct-2020 01:00 (01:00 AM) |

or in your local time:
* https://www.timeanddate.com/worldclock/fixedtime.html?msg=npm+live+feedback+triage+meeting+2020-10-27&iso=20201027T1400

## Agenda

Hand compiled by @MylesBorins 

### Package meta data

There were a number of discussions opened requesting various meta data about packages. While the majority of this is focused on the website, there is potential overlap with the registry + cli as to how we would present and collect this information.

I think we should go through the list, spend a bit of time on each specific request and make a decision if we want to do a larger epic around improving package meta data as well as the design for packages on the website. If we do let's discuss which requests would potentially get prioritized.

* [npm score](https://github.com/npm/feedback/discussions/66)
* [Add TypeScript type information to an npm package's page, and ideally in the search index](https://github.com/npm/feedback/discussions/64)
* [Show package usage by version](https://github.com/npm/feedback/discussions/60)
* [Show Bundlephobia or similar browser size stats on package pages](https://github.com/npm/feedback/discussions/65)
* [Make the repo for a package more prominent](https://github.com/npm/feedback/discussions/50)
* [Make it easier to identify the format of a package](https://github.com/npm/feedback/discussions/13)
* [Make it easier to identify system access required by a package (and changes to that access over time)](https://github.com/npm/feedback/discussions/14)

### Improvements to 2FA + publishing + tokens

Last quarter we launched npm automation tokens. We've received great feedback about ways that we can improve them. Let's discuss what we could prioritize next.

* [Naming npm tokens](https://github.com/npm/feedback/discussions/67)
* [Feedback on CI-specific auth codes](https://github.com/npm/feedback/discussions/44)
* [2FA makes publishing from continuous deployment difficult](https://github.com/npm/feedback/discussions/8)

### Frustration with CVEs and audit

There has been awesome feedback in the ecosystem related to CVE reporting as well as dark patterns we are seeing in the ecosystem that create additional work for maintainers. Let's discuss the open discussion and see if we have ideas for next steps.

* [Reduce the noise, work, and frustration from CVE reporting](https://github.com/npm/feedback/discussions/62)

### Additional Topics

We may not have time for the below discussion topics in this week's meeting. If we get through the above topics we will do a quick review of the below topics and pick the ones we want to prioritize discussion around.

#### Registry

* [npm publish finishes before publish is done](https://github.com/npm/feedback/discussions/68)
* [Document the registry endpoints](https://github.com/npm/feedback/discussions/55)

#### CLI

* [Add npm scripts as first level commands - `npm dev`](https://github.com/npm/feedback/discussions/54)
* [Make it easier to install a package from a pull request](https://github.com/npm/feedback/discussions/39)
* [Support for Browser-Based Auth from CLIs](https://github.com/npm/feedback/discussions/46)
* [make engines field usable with postinstall scripts](https://github.com/npm/feedback/discussions/58)
* [Lockfile interop between npm6 and 7](https://github.com/npm/feedback/discussions/59)
* [Support for groups of dependencies (beyond just prod/dev)](https://github.com/npm/feedback/discussions/57)
* [Workspaces in the npm CLI](https://github.com/npm/feedback/discussions/5)

#### Website

* [Use GitHub Flavored Markdown for README rendering](https://github.com/npm/feedback/discussions/3)

## Public Participation

We stream our conference call straight to YouTube so anyone can listen to it live, it should start playing at https://www.youtube.com/c/npminc/live when we turn it on. There's usually a short cat-herding time at the start of the meeting and then occasionally we have some quick private business to attend to before we can start recording & streaming. So be patient and it should show up.

If you have specific questions you would like us to answer that are outside of the planned agenda please start a discussion thread in this

**Comment by wesleytodd** [comment]
Can you add a calendar event to the npm calendar?

**Reply by MylesBorins** [reply]
@darcyclarke can you add something please?

**Reply by wesleytodd** [reply]
Sorry I missed it, I rely too much on my calendar to remind me 🤦

**Comment by Ethan-Arrowood** [comment]
the timezone table has the wrong date I believe

**Reply by Trott** [reply]
> the timezone table has the wrong date I believe

Ah, that's why I thought it was on Thursday until I saw Myles's tweet saying Tuesday.....

**Reply by MylesBorins** [reply]
Fixed! Thanks for the headsup

**Comment by MylesBorins** [comment]
# npm Public Feedback Triage

### Attendees

- Myles Borins (@mylesborins)
- Darcy Clarke (@darcyclarke)
- Edward Thomson (@ethompson)
- Meirav Feiler
- Scott Densmore
- Bryan Clark
- Ruy Adorno

### Agenda

### Package meta data

* [npm score](https://github.com/npm/feedback/discussions/66)
* [Add TypeScript type information to an npm package's page, and ideally in the search index](https://github.com/npm/feedback/discussions/64)
* [Show package usage by version](https://github.com/npm/feedback/discussions/60)
* [Show Bundlephobia or similar browser size stats on package pages](https://github.com/npm/feedback/discussions/65)
* [Make the repo for a package more prominent](https://github.com/npm/feedback/discussions/50)
* [Make it easier to identify the format of a package](https://github.com/npm/feedback/discussions/13)
* [Make it easier to identify system access required by a package (and changes to that access over time)](https://github.com/npm/feedback/discussions/14)

---

### Notes

#### npm scores

- Popularity / Quality / Mainenance data is only shown on search. We could include on a package landing page
- We should invest in making these existing metrics higher quality
- We might want to improve discoverability of being able to search by these metrics

#### TypeScript type information

- Would this information be part of the packument in the registry?
- There are other runtimes / languages. Should we privledge it? Many moons ago we could have done this for coffee script.
- Would this be something on publish via metadata or something that would be done via static analysis

#### Show package usage by version

- This would be dependent on how we do analytics
- Registry would have to do some work on how data is currently being stored, there would be some work to be done.

#### Show size stats

- Perhaps not something we should focus on initially
- Bundle size is higher effort lower reward compared to some of the other package information improvements we are talking about
- We can use bundle size to improve our metrics to some capacity
- We can link to bundlephobia and explain why we chose not to directly surface this information if we make a landing page explaining our scoring ideology

#### Make it easier to identify the format of a package

- We should perhaps start with one and add more identification over time
- Runtime vs syntax
- Identifying ESM
